ingestion/justice_data_source/source.py

```diff
@@ -54,17 +66,68 @@ def create(cls, config_dict, ctx):
         return cls(ctx, config)
 
     def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
-        # TODO generate metadata for the dashboard itself
+        all_chart_data = self.client.list_all(self.config.exclude_id_list)
+
+        # create group entities for publication owners
+        owner_emails = {chart["owner_email"] for chart in all_chart_data}
+        for owner_email in owner_emails:
+            mce = self._make_group(owner_email)
+            wu = MetadataWorkUnit("single_mce", mce=mce)
+            self.report.report_workunit(wu)
+            yield wu
 
-        for chart_data in self.client.list_all():
+        # creates each chart entity
+        for chart_data in all_chart_data:
             mce = self._make_chart(chart_data)
             wu = MetadataWorkUnit("single_mce", mce=mce)
             self.report.report_workunit(wu)
             yield wu
 
+        # adds domains to each chart created previously (if chart has a domain)
+        for chart_data in all_chart_data:
+            if chart_data.get("domain"):
+                mcp = self._assign_chart_to_domain(chart_data)
+                wu = MetadataWorkUnit("single_mcp", mcp=mcp)
+                self.report.report_workunit(wu)
+                yield wu
+
+        # make the dashboard itself
+        chart_urns = [
+            builder.make_chart_urn(self.platform_name, chart_data["id"])
+            for chart_data in all_chart_data
+        ]
+        mce = self._make_dashboard(chart_urns)
+        wu = MetadataWorkUnit("single_mce", mce=mce)
+        self.report.report_workunit(wu)
+        yield wu
+
     def get_report(self):
         return self.report
 
+    def _make_dashboard(self, chart_urns):
+        dashboard_urn = builder.make_dashboard_urn(self.platform_name, "Justice Data")
+        dashboard_snapshot = DashboardSnapshot(
+            urn=dashboard_urn,
+            aspects=[Status(removed=False)],
+        )
+
+        dashboard_info = DashboardInfoClass(
+            title="Justice Data",
```

Comment:
Believe it already has an id as specified by the urn created in a previous step - because combined into the same snapshot object it will always be associated with that urn and not duplicate